src/Microsoft.IdentityModel.Tokens/CryptoProviderFactory.cs

```diff
@@ -177,33 +178,51 @@ public virtual bool IsSupportedAlgorithm(string algorithm, SecurityKey key)
                 if (jsonWebKey.Kty == JsonWebAlgorithmsKeyTypes.RSA)
                     return IsRsaAlgorithmSupported(algorithm);
                 else if (jsonWebKey.Kty == JsonWebAlgorithmsKeyTypes.EllipticCurve)
-                    return IsEcdsaAlgorithmSupported(algorithm);
+                {
+#if NETSTANDARD1_4
+                    ECDsa ecdsa = AsymmetricSignatureProvider.CreateECDsaFromJsonWebKey(jsonWebKey, isPrivateKey);
+#else
+                    ECDsaCng ecdsa = AsymmetricSignatureProvider.CreateECDsaFromJsonWebKey(jsonWebKey, isPrivateKey);
+#endif
+                    return IsEcdsaAlgorithmSupported(algorithm, ecdsa);
+                }
                 else if (jsonWebKey.Kty == JsonWebAlgorithmsKeyTypes.Octet)
                     return IsSymmetricAlgorithmSupported(algorithm);
 
                 return false;
             }
 
-            if (key as ECDsaSecurityKey != null)
-                return IsEcdsaAlgorithmSupported(algorithm);
+            ECDsaSecurityKey ecdsaSecurityKey = key as ECDsaSecurityKey;
+            if (ecdsaSecurityKey != null)
+                return IsEcdsaAlgorithmSupported(algorithm, ecdsaSecurityKey.ECDsa);
 
             if (key as SymmetricSecurityKey != null)
                 return IsSymmetricAlgorithmSupported(algorithm);
 
             return false;
         }
 
-        private bool IsEcdsaAlgorithmSupported(string algorithm)
+        private bool IsEcdsaAlgorithmSupported(string algorithm, ECDsa key)
         {
             switch (algorithm)
             {
                 case SecurityAlgorithms.EcdsaSha256:
-                case SecurityAlgorithms.EcdsaSha384:
-                case SecurityAlgorithms.EcdsaSha512:
                 case SecurityAlgorithms.EcdsaSha256Signature:
+                    if (key.KeySize == 256)
```

Comment:
The cool thing with adding a 1.6 TFM is that it would allow us to make the ECDSA support truly cross-platform (currently, it directly deals with CNG blobs, which can't work on Linux/MacOS: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/master/src/Microsoft.IdentityModel.Tokens/AsymmetricSignatureProvider.cs#L418-L555 :sweat_smile:)